UefiPayloadPkg/PlatformCI/PlatformBuild.py

```diff
@@ -0,0 +1,224 @@
+        ''' Add command line options to the argparser '''
+        parserObj.add_argument('-a', "--arch", dest="build_arch", type=str, default="X64",
+                               help="Optional - architecture to build.  IA32 will use IA32 for Pei & Dxe. "
+                               "X64 will use X64 for both PEI and DXE.")
+
+    def RetrieveCommandLineOptions(self, args):
+        '''  Retrieve command line options from the argparser '''
+
+        shell_environment.GetBuildVars().SetValue(
+            "TARGET_ARCH", args.build_arch.upper(), "From CmdLine")
+        shell_environment.GetBuildVars().SetValue(
+            "ACTIVE_PLATFORM", "UefiPayloadPkg/UefiPayloadPkg.dsc", "From CmdLine")
+        self.args = args
+
+
+    def GetWorkspaceRoot(self):
+        ''' get WorkspacePath '''
+        return CommonPlatform.WorkspaceRoot
+
+    def GetPackagesPath(self):
+        ''' Return a list of workspace relative paths that should be mapped as edk2 PackagesPath '''
+        return ()
+
+    def GetActiveScopes(self):
+        ''' return tuple containing scopes that should be active for this process '''
+        return CommonPlatform.Scopes
+
+    def GetName(self):
+        ''' Get the name of the repo, platform, or product being build '''
+        ''' Used for naming the log file, among others '''
+
+        # check the startup nsh flag and if set then rename the log file.
+        # this helps in CI so we don't overwrite the build log since running
+        # uses the stuart_build command.
+        if(shell_environment.GetBuildVars().GetValue("MAKE_STARTUP_NSH", "FALSE") == "TRUE"):
+            return "UefiPayloadPkg_With_Run"
+        return "UefiPayloadPkg"
+
+    def GetLoggingLevel(self, loggerType):
+        ''' Get the logging level for a given type
+        base == lowest logging level supported
+        con  == Screen logging
+        txt  == plain text file logging
+        md   == markdown file logging
+        '''
+        return logging.DEBUG
+
+    def SetPlatformEnv(self):
+        logging.debug("PlatformBuilder SetPlatformEnv")
+        self.env.SetValue("PRODUCT_NAME", "UefiPayloadPkg", "Platform Hardcoded")
+        self.env.SetValue("TOOL_CHAIN_TAG", "VS2022", "Default Toolchain")
+
+        # Add support for using the correct Platform Headers, tools, and Libs based on architecture
+        # requested to be built when building VS2022 or VS2019
+        if self.env.GetValue("TOOL_CHAIN_TAG") == "VS2022" or self.env.GetValue("TOOL_CHAIN_TAG") == "VS2019":
+            key = self.env.GetValue("TOOL_CHAIN_TAG") + "_HOST"
+            if self.env.GetValue("TARGET_ARCH") == "IA32":
+                shell_environment.ShellEnvironment().set_shell_var(key, "x86")
+            elif self.env.GetValue("TARGET_ARCH") == "X64":
+                shell_environment.ShellEnvironment().set_shell_var(key, "x64")
+        shell_environment.ShellEnvironment().set_shell_var("CLANG_HOST_BIN", "n")

```

Comment:
This needs to be inside the "TOOL_CHAIN_TAG" check for VS2022, otherwise our make command becomes `nmake` for the CLANGDWARF portion of the build, even on a linux based system.

```suggestion
            shell_environment.ShellEnvironment().set_shell_var("CLANG_HOST_BIN", "n")

```

Reply:
Ideally case is windows using nmake and linux based machine using make.
change my code to use os.name == 'nt' to decide nmake or make, change my code for c + d + e + f cases.

In current BaseTools design is like below

a. Windows - Build BaseTools

MSVC + nmake
b. Linux - Build BaseTools

GCC + make
c. Windows - Build EFI module by dsc file

MSVC + nmake = VS2019 and VS2022
d. Windows - Build EFI module by dsc file

LLVM (clang) + nmake = CLANGPDB and CLANGDWARF
e. Linux - Build EFI module by dsc file

GCC + make = GCC
f. Linux - Build EFI module by dsc file

LLVM (clang) + make = CLANGPDB and CLANGDWARF

Reply:
I'm happy with your preferred solution. Thanks!